Added ru_ru.json

Felix14-v2 opened the pull request:
Yes, I know that translations should be suggested on OneSky, but the translation strings have not been updated there since 1.16, and the quality of the existing translation into Russian in places is poor. Sodium is **very technical** in describing its options, and in order to maintain the accuracy of the description, it is necessary to use the exact forms and meanings of Russian words in translation. However, people sometimes do not understand the meaning of translated terms and keep voting for incorrect translations.

I know that two pull requests with translations into Russian (as well as many other languages) have already been closed. But the quality of this translations suggested here was also not enough to approve them.

Therefore, I am creating my pull request.
Translation features:
• guiding the uniform principles (style) of translating programs from English into Russian
• translation uses already correctly translated phrases from OneSky
• translator understands each translated term
• translator is a native speaker
• amount of free space in the Sodium UI was taken into account (including in-game tests)

I understand that the development team's policy for translating the interface conflicts with a single person request approval. However, the situation with the quality of Russian translation on OneSky is really disappointing, and I am trying to fix it at least this way.

Have a nice day! And thank you for taking your time for me.

Felix14-v2 committed:
Added ru_ru.json

maityyy commented:
I don't think this PR will be accepted, but this translation is good.

VidTu commented:
По моему субъективному мнению, перевод "Buy us a coffee!" как "Подарите нам кофе!" (`sodium.options.buttons.donate`) выглядит странно. "Купите нам кофе!" или "Угостите нас кофе!" получше выглядит в контексте пожертовавания.

Felix14-v2 committed:
Some improvements (see the commit description)

• Improved compliance with the source text (gui_scale.tooltip, view_bobbing.tooltip)
• Improved localization (graphics_quality.tooltip, entity_distance.tooltip, vignette.tooltip)
• Refund from localized variant to match the official MC translation. It sounds worse, but this is the official translation ¯\_(-_-)_/¯ (leaves_quality.tooltip)
• General compliance with the official translation of "Chunk Building" in 1.18.1 (smooth_lighting.tooltip, biome_blend.tooltip, chunk_update_threads.tooltip)
• Clearer wording (mipmap_levels.tooltip)
• Avoiding tautology (use_block_face_culling.tooltip, allow_direct_memory_access.tooltip)
• A shorter phrase to fit into the GUI space (cpu_render_ahead_limit.name) (Give us more space! :D)

Felix14-v2 commented:
> По моему субъективному мнению, перевод "Buy us a coffee!" как "Подарите нам кофе!" (sodium.options.buttons.donate) выглядит странно. "Купите нам кофе!" или "Угостите нас кофе!" получше выглядит в контексте пожертвования.

<details>
 <summary>На русском</summary>
 Полностью согласен, но...
Как я и писал в своём комментарии к этой строчке на [Weblate](https://translate.eclixo.com/translate/caffeinemc/sodium/ru/?q=state%3A%3Ctranslated&offset=2#comments), прямой перевод звучит слишком категорично. В русском языке слово "купите" воспринимается иначе, чем его прямой эквивалент "buy" в английском. Вариант с "подарите" устраняет своеобразное требование действия со стороны пользователя, при этом сохраняется призыв к поддержке. Вариант "Угостите нас кофе", пожалуй, даже более лучший, однако он меняет вид перевода с "частично локализованный" до "полностью локализованный". Думаю, что решение по локализации вместо прямого перевода не может быть принято в одиночку, поэтому обсуждение перевода только приветствуется. Что вы думаете об этом?

Также рекомендую добавить ваш вариант на Weblate. В отличие от OneSky он подгружает строки перевода напрямую с GitHub, поэтому текст там всегда будет актуальным. Также можете посетить [ветку обсуждения переводов](https://discord.com/channels/602796788608401408/922250280748335114) для Sodium в Discord.
</details>

I totally agree, but...
As I wrote in my comment to this line on [Weblate](https://translate.eclixo.com/translate/caffeinemc/sodium/ru/?q=state%3A%3Ctranslated&offset=2#comments), the direct translation sounds too categorical. In Russian, the word "купите" is perceived differently than its direct equivalent "buy" in English. The word "подарите" eliminates the implicit requirement for action on the part of the user, while maintaining a call for support. The "Угостите нас кофе" option is perhaps even better, but it changes the type of translation from "partially localized" to "fully localized". I think that the decision to localize instead of direct translation cannot be made alone, so discussion of translation is welcome. What do you think about it?

I also recommend adding your version to Weblate. Unlike OneSky, it loads translation strings directly from GitHub, so the text there will always be relevant. You can also visit the [discussion thread of translations](https://discord.com/channels/602796788608401408/922250280748335114) for Sodium in Discord.

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.always_defer_chunk_updates.tooltip": "Если включено, рендеринг никогда не будет ожидать завершения обновлений чанков, даже если они важны. Это способствует сильному увеличению частоты кадров в некоторых ситуациях, но может привести к появлению существенных визуальных ошибок в мире.",
+  "sodium.options.buttons.undo": "Отмена",
+  "sodium.options.buttons.apply": "Применить",
+  "sodium.options.buttons.donate": "Подарите нам кофе!"
```
```suggestion
  "sodium.options.buttons.donate": "Угостите нас кофе!"
```

Altegar replied:
Как по мне, "Угостите нас кофе!" лучше подходит, чем "Купите" или "Подарите" нам кофе

Felix14-v2 replied:
Подождём ещё чей-то голос и если вопросов не будет – применим изменения.

UPD: на почту пришло письмо с твоим комментарием по поводу виньетки, но тут я его почему-то не нашёл. Искал плохо, наверное, ну да не суть, отвечу здесь. Виньетка – это рамка вокруг фото, а виньетирование – оптический эффект затенения краёв изображения. Вот [виньетка](https://catherineasquithgallery.com/uploads/posts/2021-02/1614256890_52-p-ramki-cherno-belie-bez-fona-54.png), а вот [виньетирование](https://funart.pro/uploads/posts/2020-04/1585772018_4-p-zatemnennie-foni-26.png). Их много где путают, но подобная ошибка в переводе недопустима.

Altegar replied:
Понял, буду знать

PinkGoosik replied:
Мне нравится "угостите нас"

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.use_persistent_mapping.name": "Использовать постоянный буфер",
+  "sodium.options.use_persistent_mapping.tooltip": "Если включено, небольшое количество памяти будет постоянно использоваться в качестве промежуточного буфера для загрузки чанков, помогая уменьшить перегрузку ЦП и нестабильность времени отрисовки кадра при загрузке/выгрузке чанков.\n\nТребуется OpenGL 4.4 или ARB_buffer_storage.",
+  "sodium.options.chunk_memory_allocator.name": "Распределитель памяти",
+  "sodium.options.chunk_memory_allocator.tooltip": "Выбирает распределитель памяти, используемый для рендеринга чанков.\n- ASYNC: Быстрейшая опция, хорошо работает с большинством современных графических драйверов.\n- SWAP: Запасная опция для старых графических драйверов. Может существенно увеличить использование памяти.",
```
Всё-таки "вариант" звучит лучше, чем "опция".

```suggestion
  "sodium.options.chunk_memory_allocator.tooltip": "Выбирает распределитель памяти, используемый для рендеринга чанков.\n- ASYNC: Быстрейший вариант, хорошо работает с большинством современных графических драйверов.\n- SWAP: Запасной вариант для старых графических драйверов. Может существенно увеличить использование памяти.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.cpu_render_ahead_limit.tooltip": "Задаёт максимальное количество кадров, которые ЦП может подготавливать для рендеринга на ГП. Очень низкие или высокие значения могут дестабилизировать частоту кадров.",
+  "sodium.options.cpu_render_ahead_limit.value": "%s кадр(ов)",
+  "sodium.options.allow_direct_memory_access.name": "Разрешить прямой доступ к памяти",
+  "sodium.options.allow_direct_memory_access.tooltip": "Если включено, некоторые критические пути кода получат разрешение использовать прямой доступ к памяти ради производительности. Зачастую это сильно снижает перегрузку ЦП при рендеринге чанков и сущностей, но может усложнить диагностику некоторых ошибок и вылетов. Отключать это стоит только если вы следуете инструкциям или точно знаете, что де ...
```
Более хорошая формулировка.

```suggestion
  "sodium.options.allow_direct_memory_access.tooltip": "Если включено, некоторым критическим путям кода будет разрешено использовать прямой доступ к памяти для повышения производительности. Зачастую это сильно снижает перегрузку ЦП при рендеринге чанков и сущностей, но может усложнить диагностику некоторых ошибок и вылетов. Отключать это стоит только если вы следуете инструкциям или точно знаете, что делаете.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.brightness.tooltip": "Управляет яркостью (гаммой) игры.",
+  "sodium.options.gui_scale.tooltip": "Устанавливает коэффициент масштабирования для интерфейса. При выборе \"авто\" всегда будет использоваться наибольший размер.",
+  "sodium.options.fullscreen.tooltip": "Если включено, игра будет открыта во весь экран (если доступно).",
+  "sodium.options.v_sync.tooltip": "Если включена, частота кадров будет синхронизирована с частотой обновления монитора, обеспечивая более плавную игру за счет небольшой задержки ввода. Возможно понижение производительности, если ваша система слишком медленна.",
```
Хорошее, но слишком сильное отклонение от исходного текста, чтобы принять его без обсуждения. Обязательно напишите свои мысли по этому поводу.

```suggestion
  "sodium.options.v_sync.tooltip": "Если включена, частота кадров будет синхронизирована с частотой обновления монитора, обеспечивая более плавную игру за счет небольшой задержки ввода. Может понизить производительность слабых систем.",
```

Altegar replied:
Кратко и понятно

Felix14-v2 reviewed (commented):
<details>
 <summary>In English</summary>

 Some improvements from the translation of @Sushomeister :)
It is advisable to discuss each of them, feel free to write your thoughts about these translations and offer your own. The more discussions there are, the more accurate and better the final translation will be.
</details>

Некоторые улучшения, нагло взятые из [перевода](https://github.com/CaffeineMC/sodium-fabric/pull/1121) @Sushomeister :)
Желательно обсудить каждое из них, не стесняйтесь писать свои мысли по поводу переводов и предлагать свои варианты. Чем больше ведётся обсуждений, тем точнее и качественнее будет итоговый перевод.

Sushomeister commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.biome_blend.tooltip": "Управляет диапазоном анализа биомов для расцветки блока. Более высокие значения сильно увеличивают затрачиваемое на сборку чанков время ради незначительного улучшения качества.",
+  "sodium.options.entity_distance.tooltip": "Управляет дальностью видимости сущностей. Высокие значения увеличивают расстояние отображения за счёт частоты кадров.",
+  "sodium.options.entity_shadows.tooltip": "Если включено, простые тени будут отображаться под мобами и другими сущностями.",
+  "sodium.options.vignette.name": "Виньетирование",
```
Можно было написать просто Виньетка

Felix14-v2 committed:
Update src/main/resources/assets/sodium/lang/ru_ru.json

Felix14-v2 committed:
Update src/main/resources/assets/sodium/lang/ru_ru.json

Felix14-v2 committed:
Update src/main/resources/assets/sodium/lang/ru_ru.json

Felix14-v2 committed:
Update src/main/resources/assets/sodium/lang/ru_ru.json

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.gui_scale.tooltip": "Устанавливает коэффициент масштабирования для интерфейса. При выборе \"авто\" всегда будет использоваться наибольший размер.",
+  "sodium.options.fullscreen.tooltip": "Если включено, игра будет открыта во весь экран (если доступно).",
+  "sodium.options.v_sync.tooltip": "Если включена, частота кадров будет синхронизирована с частотой обновления монитора, обеспечивая более плавную игру за счет небольшой задержки ввода. Может понизить производительность слабых систем.",
+  "sodium.options.fps_limit.tooltip": "Ограничивает максимальное количество кадров в секунду. Это поможет уменьшить расход заряда батареи и нагрузку на систему в обеспечении многозадачности. При включённой вертикальной синхронизации этот параметр будет игнорироваться, если он выше частоты обновления вашего экрана.",
```
Появился вариант за счёт отклонения от исходного текста немного сократить длину перевода, при этом улучшив восприятие:

```suggestion
  "sodium.options.fps_limit.tooltip": "Ограничивает максимальное количество кадров в секунду. Это поможет уменьшить энергопотребление и нагрузку на систему в обеспечении многозадачности. При включённой вертикальной синхронизации этот параметр будет игнорироваться, если он выше частоты обновления вашего экрана.",
```
Что думаете?
Лично я очень не люблю отступать от исходных формулировок, хотя это тот случай, когда перевод точнее описывает ситуацию, чем исходная строка.

Altegar replied:
Мне нравится

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.use_entity_culling.name": "Отбраковка сущностей",
+  "sodium.options.use_entity_culling.tooltip": "Если включено, сущности, обнаруженные вне видимых чанков, не будут выводиться на рендеринг. Это может помочь улучшить производительность, исключив рендеринг сущностей, находящихся под землёй или за стенами.",
+  "sodium.options.use_particle_culling.name": "Фильтрация частиц",
+  "sodium.options.use_particle_culling.tooltip": "Если включено, будут отображаться только те частицы, которые определены как видимые. Это может обеспечить значительное улучшение частоты кадров, когда много частиц находится рядом.",
```
Немного переосмыслил алгоритм работы фильтрации и понял, что уместнее будет сделать акцент на (подразумеваемом) положении камеры, рядом с которым находится много частиц, а не на близком положении частиц относительно друг друга. Достигается простой перестановкой слов.

```suggestion
  "sodium.options.use_particle_culling.tooltip": "Если включено, будут отображаться только те частицы, которые определены как видимые. Это может обеспечить значительное улучшение частоты кадров, когда рядом находится много частиц.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.chunk_memory_allocator.async": "Async",
+  "sodium.options.chunk_memory_allocator.swap": "Swap",
+  "sodium.options.chunk_update_threads.name": "Потоки обновления чанков",
+  "sodium.options.chunk_update_threads.tooltip": "Задаёт количество потоков для сборки чанков. Использование большего количества потоков способствует повышению скорости обновления и загрузки чанков, но может негативно повлиять на время отрисовки.",
```
Судя по устоявшейся терминологии, "Frame time" следует переводить как "время кадра".
Как по мне, предыдущий вариант был лучше из-за бо́льшей понятности для новичков, но думаю, что так будет правильнее.

```suggestion
  "sodium.options.chunk_update_threads.tooltip": "Задаёт количество потоков для сборки чанков. Использование большего количества потоков способствует повышению скорости обновления и загрузки чанков, но может негативно повлиять на время кадра.",
```

Felix14-v2 reviewed (commented):
Ещё парочка малозначимых улучшений (я надеюсь). Не стесняйтесь высказывать мнение насчёт изменений.

Felix14-v2 committed:
Update src/main/resources/assets/sodium/lang/ru_ru.json

Felix14-v2 committed:
Update src/main/resources/assets/sodium/lang/ru_ru.json

Felix14-v2 committed:
Update src/main/resources/assets/sodium/lang/ru_ru.json

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.view_distance.tooltip": "Дальность прорисовки определяет, как далеко будет видна местность. Меньшие значения соответствуют меньшей площади, что улучшает частоту кадров.",
+  "sodium.options.simulation_distance.tooltip": "Дальность симуляции определяет, как далеко местность и сущности будут загружаться и обрабатываться. Меньшие значения могут уменьшить нагрузку на внутренний сервер и увеличить частоту кадров.",
+  "sodium.options.brightness.tooltip": "Управляет яркостью (гаммой) игры.",
+  "sodium.options.gui_scale.tooltip": "Устанавливает коэффициент масштабирования для интерфейса. При выборе \"авто\" всегда будет использоваться наибольший размер.",
```
Better consistency with the official localization
```suggestion
  "sodium.options.gui_scale.tooltip": "Устанавливает коэффициент масштабирования для интерфейса. При выборе «Авто» всегда будет использоваться наибольший размер.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.view_bobbing.tooltip": "Если включено, обзор игрока будет покачиваться при движении. Отключение этого параметра может помочь людям, страдающим от укачивания во время игры.",
+  "sodium.options.attack_indicator.tooltip": "Управляет расположением индикатора атаки на экране.",
+  "sodium.options.autosave_indicator.tooltip": "Если включено, при сохранении игрой мира на диск будет показываться индикатор.",
+  "sodium.options.graphics_quality.tooltip": "Качество графики по умолчанию управляет некоторыми наследующими настройками и необходимо для совместимости с модами. Опции ниже со значением \"По умолчанию\" будут использовать этот параметр.",
```
Better consistency wth the official localization and improved connectivity with the legacy options
```suggestion
  "sodium.options.graphics_quality.tooltip": "Качество графики по умолчанию управляет некоторыми зависимыми настройками и необходимо для совместимости с модами. Опции ниже со значением «По умолчанию» будут использовать этот параметр.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.leaves_quality.name": "Качество листьев",
+  "sodium.options.leaves_quality.tooltip": "Управляет качеством листьев.",
+  "sodium.options.particle_quality.name": "Качество частиц",
+  "sodium.options.particle_quality.tooltip": "Управляет количеством частиц, которые могут присутствовать на экране в любой момент времени.",
```
Improvement from Crowdin: the same meaning, but with better sounding
```suggestion
  "sodium.options.particle_quality.tooltip": "Управляет максимальным количеством частиц, которые могут одновременно присутствовать на экране.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.particle_quality.name": "Качество частиц",
+  "sodium.options.particle_quality.tooltip": "Управляет количеством частиц, которые могут присутствовать на экране в любой момент времени.",
+  "sodium.options.smooth_lighting.tooltip": "Управляет мягким освещением и затенением блоков. Это немного увеличивает время сборки чанка, однако не влияет на частоту кадров.",
+  "sodium.options.biome_blend.value": "%s блок(ов)",
```
```suggestion
  "sodium.options.biome_blend.value": "Блоков: %s",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.biome_blend.value": "%s блок(ов)",
+  "sodium.options.biome_blend.tooltip": "Управляет диапазоном анализа биомов для расцветки блока. Более высокие значения сильно увеличивают затрачиваемое на сборку чанков время ради незначительного улучшения качества.",
+  "sodium.options.entity_distance.tooltip": "Управляет дальностью видимости сущностей. Высокие значения увеличивают расстояние отображения за счёт частоты кадров.",
+  "sodium.options.entity_shadows.tooltip": "Если включено, простые тени будут отображаться под мобами и другими сущностями.",
```
The same words and meaning, just a better formulation
```suggestion
  "sodium.options.entity_shadows.tooltip": "Если включено, под мобами и другими сущностями будут отображаться простые тени.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.vignette.tooltip": "Если включено, на экране игрока будет отображаться эффект виньетирования. Вряд ли повлияет на частоту кадров, если ваш ГП не ограничен скоростью заполнения пикселей.",
+  "sodium.options.mipmap_levels.tooltip": "Управляет количеством уровней детализации для текстур блоков. Более высокие значения обеспечивают лучший вид на расстоянии, но могут негативно повлиять на производительность с большим количеством анимированных текстур.",
+  "sodium.options.use_block_face_culling.name": "Отбраковка поверхностей блоков",
+  "sodium.options.use_block_face_culling.tooltip": "Если включено, на рендеринг будут выводиться только видимые для камеры поверхности блоков. Это может исключить большое количество поверхностей на ранней стадии отрисовки, экономя пропускную способность памяти и время для ГП. Некоторые пакеты ресурсов могут иметь проблемы с этой опцией, поэтому попробуйте отключить её, если вы видите дыры в блока ...
```
Better consistency wth the official localization
```suggestion
  "sodium.options.use_block_face_culling.tooltip": "Если включено, на рендеринг будут выводиться только видимые для камеры поверхности блоков. Это может исключить большое количество поверхностей на ранней стадии отрисовки, экономя пропускную способность памяти и время для ГП. Некоторые наборы ресурсов могут иметь проблемы с этой опцией, поэтому попробуйте отключить её, если вы видите дыры в блоках.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.use_block_face_culling.name": "Отбраковка поверхностей блоков",
+  "sodium.options.use_block_face_culling.tooltip": "Если включено, на рендеринг будут выводиться только видимые для камеры поверхности блоков. Это может исключить большое количество поверхностей на ранней стадии отрисовки, экономя пропускную способность памяти и время для ГП. Некоторые пакеты ресурсов могут иметь проблемы с этой опцией, поэтому попробуйте отключить её, если вы видите дыры в блока ...
+  "sodium.options.use_fog_occlusion.name": "Окклюзия тумана",
+  "sodium.options.use_fog_occlusion.tooltip": "Если включено, чанки, определённые как полностью скрытые в тумане, не будут отображаться ради улучшения производительности. Улучшение может быть более значительным, когда эффект тумана сильнее (например, под водой), но в некоторых сценах это может привести к нежелательным визуальным артефактам между небом и туманом.",
```
Better sounding, this word is used more often
```suggestion
  "sodium.options.use_fog_occlusion.tooltip": "Если включено, чанки, определённые как полностью скрытые в тумане, не будут отображаться ради повышения производительности. Улучшение может быть более значительным, когда эффект тумана сильнее (например, под водой), но в некоторых сценах это может привести к нежелательным визуальным артефактам между небом и туманом.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.use_fog_occlusion.tooltip": "Если включено, чанки, определённые как полностью скрытые в тумане, не будут отображаться ради улучшения производительности. Улучшение может быть более значительным, когда эффект тумана сильнее (например, под водой), но в некоторых сценах это может привести к нежелательным визуальным артефактам между небом и туманом.",
+  "sodium.options.use_entity_culling.name": "Отбраковка сущностей",
+  "sodium.options.use_entity_culling.tooltip": "Если включено, сущности, обнаруженные вне видимых чанков, не будут выводиться на рендеринг. Это может помочь улучшить производительность, исключив рендеринг сущностей, находящихся под землёй или за стенами.",
+  "sodium.options.use_particle_culling.name": "Фильтрация частиц",
```
Fix from Crowdin
```suggestion
  "sodium.options.use_particle_culling.name": "Отбраковка частиц",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.use_particle_culling.name": "Фильтрация частиц",
+  "sodium.options.use_particle_culling.tooltip": "Если включено, будут отображаться только те частицы, которые определены как видимые. Это может обеспечить значительное улучшение частоты кадров, когда рядом находится много частиц.",
+  "sodium.options.animate_only_visible_textures.name": "Анимировать только видимое",
+  "sodium.options.animate_only_visible_textures.tooltip": "Если включено, будут обновляться лишь видимые анимированные текстуры. Это может значительно повысить частоту кадров на некоторых устройствах, особенно с \"тяжёлыми\" пакетами ресурсов. Если у каких-то текстур возникают проблемы с анимацией, попробуйте отключить эту опцию.",
```
Better consistency wth the official localization
```suggestion
  "sodium.options.animate_only_visible_textures.tooltip": "Если включено, будут обновляться лишь видимые анимированные текстуры. Это может значительно повысить частоту кадров на некоторых устройствах, особенно с \"тяжёлыми\" наборами ресурсов. Если у каких-то текстур возникают проблемы с анимацией, попробуйте отключить эту опцию.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.use_particle_culling.tooltip": "Если включено, будут отображаться только те частицы, которые определены как видимые. Это может обеспечить значительное улучшение частоты кадров, когда рядом находится много частиц.",
+  "sodium.options.animate_only_visible_textures.name": "Анимировать только видимое",
+  "sodium.options.animate_only_visible_textures.tooltip": "Если включено, будут обновляться лишь видимые анимированные текстуры. Это может значительно повысить частоту кадров на некоторых устройствах, особенно с \"тяжёлыми\" пакетами ресурсов. Если у каких-то текстур возникают проблемы с анимацией, попробуйте отключить эту опцию.",
+  "sodium.options.cpu_render_ahead_limit.name": "Лимит пре-рендеринга",
```
I guess, this is the best translation. Also, an extended version of this phrase is used in the Nvidia Driver Control Panel
```suggestion
  "sodium.options.cpu_render_ahead_limit.name": "Лимит подготовки кадров",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.animate_only_visible_textures.tooltip": "Если включено, будут обновляться лишь видимые анимированные текстуры. Это может значительно повысить частоту кадров на некоторых устройствах, особенно с \"тяжёлыми\" пакетами ресурсов. Если у каких-то текстур возникают проблемы с анимацией, попробуйте отключить эту опцию.",
+  "sodium.options.cpu_render_ahead_limit.name": "Лимит пре-рендеринга",
+  "sodium.options.cpu_render_ahead_limit.tooltip": "Задаёт максимальное количество кадров, которые ЦП может подготавливать для рендеринга на ГП. Очень низкие или высокие значения могут дестабилизировать частоту кадров.",
+  "sodium.options.cpu_render_ahead_limit.value": "%s кадр(ов)",
```
The same trick as in the 28th line - we avoid inconsistencies with numerals, while maintaining a good appearance
```suggestion
  "sodium.options.cpu_render_ahead_limit.value": "Кадров: %s",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.cpu_render_ahead_limit.tooltip": "Задаёт максимальное количество кадров, которые ЦП может подготавливать для рендеринга на ГП. Очень низкие или высокие значения могут дестабилизировать частоту кадров.",
+  "sodium.options.cpu_render_ahead_limit.value": "%s кадр(ов)",
+  "sodium.options.allow_direct_memory_access.name": "Разрешить прямой доступ к памяти",
+  "sodium.options.allow_direct_memory_access.tooltip": "Если включено, некоторым критическим путям кода будет разрешено использовать прямой доступ к памяти для повышения производительности. Зачастую это сильно снижает перегрузку ЦП при рендеринге чанков и сущностей, но может усложнить диагностику некоторых ошибок и вылетов. Отключать это стоит только если вы следуете инструкциям или точно знаете, ...
```
A more correct and well-sounding option. Also, fixed missing comma
```suggestion
  "sodium.options.allow_direct_memory_access.tooltip": "Если включено, некоторым важнейшим частям кода будет разрешено использовать прямой доступ к памяти для повышения производительности. Зачастую это сильно снижает накладные расходы на ЦП при рендеринге чанков и сущностей, но может усложнить диагностику некоторых ошибок и вылетов. Отключать это стоит, только если вы следуете инструкциям или точно знаете, что делаете.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.enable_memory_tracing.name": "Включить трассировку памяти",
+  "sodium.options.enable_memory_tracing.tooltip": "Функция отладки. Если включено, трассировка стека будет собираться вместе с распределением памяти, чтобы помочь улучшить диагностическую информацию при обнаружении утечек памяти.",
+  "sodium.options.performance_impact_string": "Влияние на производительность: %s",
+  "sodium.options.use_persistent_mapping.name": "Использовать постоянный буфер",
```
A more specific formulation, also now this option has the same semantic structure as the others in the same tab
```suggestion
  "sodium.options.use_persistent_mapping.name": "Постоянный буфер памяти",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.enable_memory_tracing.tooltip": "Функция отладки. Если включено, трассировка стека будет собираться вместе с распределением памяти, чтобы помочь улучшить диагностическую информацию при обнаружении утечек памяти.",
+  "sodium.options.performance_impact_string": "Влияние на производительность: %s",
+  "sodium.options.use_persistent_mapping.name": "Использовать постоянный буфер",
+  "sodium.options.use_persistent_mapping.tooltip": "Если включено, небольшое количество памяти будет постоянно использоваться в качестве промежуточного буфера для загрузки чанков, помогая уменьшить перегрузку ЦП и нестабильность времени отрисовки кадра при загрузке/выгрузке чанков.\n\nТребуется OpenGL 4.4 или ARB_buffer_storage.",
```
More correct and well-sounding option
```suggestion
  "sodium.options.use_persistent_mapping.tooltip": "Если включено, небольшое количество памяти будет постоянно использоваться в качестве промежуточного буфера для загрузки чанков, помогая уменьшить накладные расходы на ЦП и нестабильность времени отрисовки кадра при их загрузке/выгрузке.\n\nТребуется OpenGL 4.4 или ARB_buffer_storage.",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.performance_impact_string": "Влияние на производительность: %s",
+  "sodium.options.use_persistent_mapping.name": "Использовать постоянный буфер",
+  "sodium.options.use_persistent_mapping.tooltip": "Если включено, небольшое количество памяти будет постоянно использоваться в качестве промежуточного буфера для загрузки чанков, помогая уменьшить перегрузку ЦП и нестабильность времени отрисовки кадра при загрузке/выгрузке чанков.\n\nТребуется OpenGL 4.4 или ARB_buffer_storage.",
+  "sodium.options.chunk_memory_allocator.name": "Распределитель памяти",
```
Avoid tautology with the description, just another form of the word
```suggestion
  "sodium.options.chunk_memory_allocator.name": "Распределение памяти",
```

Felix14-v2 commented on src/main/resources/assets/sodium/lang/ru_ru.json:
```diff
@@ -0,0 +1,66 @@
+  "sodium.options.always_defer_chunk_updates.tooltip": "Если включено, рендеринг никогда не будет ожидать завершения обновлений чанков, даже если они важны. Это способствует сильному увеличению частоты кадров в некоторых ситуациях, но может привести к появлению существенных визуальных ошибок в мире.",
+  "sodium.options.buttons.undo": "Отмена",
+  "sodium.options.buttons.apply": "Применить",
+  "sodium.options.buttons.donate": "Угостите нас кофе!"
```
Translations for ModMenu description and summary (a shortened version of the description that can fit into a limited space)
```suggestion
  "sodium.options.buttons.donate": "Угостите нас кофе!",
   "modmenu.summaryTranslation.sodium": "Свободный и открытый мод для оптимизации Minecraft.",
   "modmenu.descriptionTranslation.sodium": "Свободный мод для оптимизации Minecraft с открытым исходным кодом, призванный уменьшить зависания и улучшить частоту кадров."
```

Felix14-v2 committed:
Apply suggestions

Discussed with other users in VK, looks fine

Felix14-v2 commented:
Для новых читателей: дальнейшее развитие инфраструктуры локализации производится в [репозитории](https://github.com/Madis0/sodium-fabric-translations) Madis0, а сами переводы следует предлагать [в соответствующем разделе](https://crowdin.com/project/sodium-fabric) Crowdin. Пока локализации не будут включены в Sodium, вы можете [скачать](https://github.com/Madis0/sodium-fabric-translations/releases/latest/download/Sodium.Translations.zip) и установить актуальную версию уже имеющихся в виде набора ресурсов.
